src/containers/AccountModal/forms/Registration.tsx

```diff
@@ -57,9 +57,11 @@ const Registration = () => {
         return;
       }
 
-      await register(email, password);
+      const cleanConsentValues = Object.fromEntries(Object.entries(consentValues).filter(isNotEmptyStringEntry).map(formatCrfEntry));
 
-      await updateConsents(customerConsents).catch(() => {
+      await register(email, password, cleanConsentValues);
+
+      await updateConsents(customerConsents, cleanConsentValues).catch(() => {
```

Comment:
Isn't this redundant for the JWP service since consents are set with the `register` call? Since there is only 1 use of register, I think it would make sense to call `updateConsents` within the implementation of `register` for the cleeng service that needs this separately. Then you can maintain a common interface and avoid redundant calls.

Reply:
Yeah, it's gotten redundant as bugs got discovered and as I apply fixes, but it is the way it's currently implemented and refactoring it would be out of the scope of this PR. I suggest leaving this for a separate task and move this PR forwards if everything else is fine.